### DISCUSSION TITLE
What's a good way to test the prompt from the codebase directly?

### DISCUSSION THREAD

**Discussion by mehcode** [starship/starship#2061] [Q&A] [answered]
When I run `./target/debug/starship prompt` it looks like:

```

%{%}starship%{%} on %{%} %{%}%{%}master%{%} is %{%}📦 %{%}%{%}v0.48.0%{%} via %{%}🦀 %{%}%{%}v1.49.0%{%}
%{%}❯%{%} %
```

Note that this works fine when initialized and rendered as the prompt. 

Is there interest in perhaps a `--plain` mode that turns off this escaping to make it easier to tinker on the prompt?

**Comment by mickimnet** [comment]
What shell are you using and did you set the environment variable `STARSHIP_SHELL`? It did work for me, when I was tinkering with my config.

Best way is to run `starship init $YOURSHELL --print-full-init` and than use that output for your tests.

**Reply by mehcode** [reply]
I'm using zsh. I tried setting the env var to zsh directly with no change. Note that using the init does work fine, I just thought the prompt command would be a bit simpler to play with.

**Reply by mickimnet** [reply]
Did you already include `eval "$(starship init zsh)"` in your `~/.zshrc`? I didn‘t do my testing during theming in _zsh_.

**Reply by mehcode** [reply]
I'm talking about editing code in the repository, doing `cargo build`, and running `./target/debug/starship`. 

My `~/.zshrc` has my globally installed version of starship running fine with the command you mention.

**Comment by davidkna** [comment, marked as answer]
For testing use something like `STARSHIP_SHELL= ./target/debug/starship prompt` works well without intefering with the regular prompt.

**Reply by andytom** [reply]
In addition to what @davidkna has suggested you can also run `starship module MODULE_NAME` to display the output of a single module (this is not shell escaped).

Side note if this is not already it might be good to add this to the `CONTRIBUTING.md` to make it easier to in the future.
